packages/flutter/lib/src/services/text_input.dart

```diff
@@ -2770,9 +2783,18 @@ class SystemContextMenuController with SystemContextMenuClient, Diagnosticable {
 
     ServicesBinding.systemContextMenuClient = this;
 
+    _customActionCallbacks.clear();
```

Comment:
I think you also might need to call this in handleSystemHide. Then I think we will be covered and this shouldn't leak.

Reply:
Also test this: Show the SystemContextMenu with a certain item, then show again with a new item. Then try sending a handleCustomContextMenuAction for the original item.